quickwit/quickwit-query/src/query_ast/tantivy_query_ast.rs

```diff
@@ -188,6 +201,72 @@ impl TantivyBoolQuery {
             // This is just a convention mimicking Elastic/Commonsearch's behavior.
             return TantivyQueryAst::match_all();
         }
+
+        let mut new_must = Vec::with_capacity(self.must.len());
+        for must in self.must {
+            let mut must_bool = match must {
+                TantivyQueryAst::Bool(bool_query) => bool_query,
+                _ => {
+                    new_must.push(must);
+                    continue;
+                }
+            };
+            if must_bool.should.is_empty() {
+                new_must.append(&mut must_bool.must);
+                self.filter.append(&mut must_bool.filter);
+                self.must_not.append(&mut must_bool.must_not);
```

Comment:
shouldn't we call simplify here, so the optimization is recursive?

Reply:
`simplify_asts` on everything at the start of the function